File: Level-2/44.java
import java.util.*;

class Solution {
    public HashSet<Integer> numberSet = new HashSet<>();
    public int solution(String numbers) {
        int answer = 0;
        // 숫자 조합 구하기
        recursive("", numbers);
        
        Iterator<Integer> it = numberSet.iterator();
        while(it.hasNext()) {
            int number = it.next();
            // 소수인지 확인
            if(isPrime(number)) {
                answer++;
            }
        }
        
        return answer;
    }
    
    public void recursive(String comb, String others) {
        if(!comb.equals("")){
            numberSet.add(Integer.valueOf(comb));
        }
        
        for(int i = 0; i < others.length(); i++) {
            recursive(comb + others.charAt(i), others.substring(0, i) + others.substring(i + 1));
        }
    }
    
    public boolean isPrime(int num) {
        // 0이나 1은 소수가 아님
        if(num == 0 || num == 1) {
            return false;
        }
        
        // 에라토스테네스의 체
        int lim = (int) Math.sqrt(num);
        for(int i = 2; i <= lim; i++) {
            if(num % i == 0) {
                return false;
            }
        }
        
        return true;
    }
}

File: Level-2/44.js
function solution(name) {
    let answer = 0;
    let minMove = name.length - 1;
    
    [...name].map((v, i) => {
        answer += Math.min(v.charCodeAt() - 65, 91 - v.charCodeAt());
        let next = i + 1;
        
        while(next < name.length && name[next] === 'A') {
            next++;
        }
        
        // BBBAAAAAABA인 경우는 다시 왼쪽으로 돌아가는 것이 더 빠르다.
        // 처음 위치로 돌아간 횟수(i*2) + 문자열길이 - A가 연속으로 나오는 지점의 다음값
        const toBack = name.length - next;
        // 오른쪽으로 가기, 왼쪽으로 가기, 뒤의 요소 먼저 입력
        minMove = Math.min(minMove, (i * 2) + toBack, i + (2 * toBack));
    });
    
    return answer + minMove;
}
